ledger/ledger-common/src/main/kotlin/net/corda/v5/ledger/common/transaction/TransactionSignatureService.kt

```diff
@@ -0,0 +1,57 @@
+package net.corda.v5.ledger.common.transaction
+
+import net.corda.v5.application.crypto.DigitalSignatureAndMetadata
+import net.corda.v5.application.crypto.DigitalSignatureVerificationService
+import net.corda.v5.base.annotations.Suspendable
+import net.corda.v5.crypto.SecureHash
+import net.corda.v5.crypto.merkle.MerkleProof
+import java.security.PublicKey
+
+/**
+ * TransactionSignatureService can be used to sign and verify transaction signatures.
+ * It supports both single and batch signatures.
+ */
+
+interface TransactionSignatureService {
+    /**
+     * Signs a transaction id with all the available keys.
+     *
+     * @param transactionId The transaction id to be signed.
+     * @param publicKeys The keys which should be tried to sign with.
```

Comment:
```suggestion
     * @param publicKeys Public keys that correspond to the private keys which should be used to sign with.
```
Is this what we meant?

Reply:
Nearly.
I wanted to highlight that it only tries to sign with those keys. It is possible that only a few of them available to sign with.
I modified your suggestion a bit to this:
```Public keys that correspond to the private keys which should be attempted to sign with.```